pkg/store/tsdb_test.go

```diff
@@ -585,7 +586,7 @@ func benchTSDBStoreSeries(t testutil.TB, totalSamples, totalSeries int) {
 
 	for j := 0; j < 3; j++ {
 		head, created := storetestutil.CreateHeadWithSeries(t, j, storetestutil.HeadGenOptions{
-			TSDBDir:          tmpDir,
+			TSDBDir:          t.TempDir(),
```

Comment:
`tmpDir` is already `t.TempDir()`, why this change?

Reply:
Oh, subsequent calls to "t.Tempdir" return different directories; is that a bug?

Reply:
I using the same local dir caused the tests to fail after upgrading to 2.46, but it does not seem to be the case. It was rather because we close the head and entire db in a non-deterministic manner by using `t.CleanUp` for one, and `defer` for the other.